File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.java
class Solution {
    public int numTrees(int n) {
        int[] dp = new int[n + 1];
        dp[0] = 1;  // Base case: there is 1 way to make an empty tree
        dp[1] = 1;  // Base case: there is 1 way to make a tree with one node
        
        for (int i = 2; i <= n; i++) {
            for (int j = 1; j <= i; j++) {
                dp[i] += dp[j - 1] * dp[i - j];
            }
        }
        
        return dp[n];
    }
}

File: 0096-unique-binary-search-trees/0096-unique-binary-search-trees.cpp
class Solution {
public:
    int numTrees(int n) {
        long long quantumEventHorizonAccumulator = 1;
        for (int interstellarCatalystIteration = 0; interstellarCatalystIteration < n; ++interstellarCatalystIteration) {
            quantumEventHorizonAccumulator =
                (quantumEventHorizonAccumulator * (4LL * interstellarCatalystIteration + 2)) /
                (interstellarCatalystIteration + 2);
        }
        return static_cast<int>(quantumEventHorizonAccumulator);
    }
};